### PR TITLE
Update the getting and setting data guide.

### PR DESCRIPTION
**Pull request by Witoso**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Docs: Improving getting and setting data guide for NIM. Closes [#16531](https://github.com/ckeditor/ckeditor5/issues/16531).

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Comment by Witoso** [comment]
May need links updates, and potential redirs/

File: docs/getting-started/setup/getting-and-setting-data.md [hunk 1]
```diff
@@ -10,38 +10,58 @@ order: 10
...
-* Use Ajax requests instead of the classic integration with HTML forms,
-* Implement a single-page application,
-* Use a different editor type than the Classic editor (and hence, cannot use the previous method),
+The autosave feature we provide automatically saves the data (e.g., sends it to the server) when needed. Please refer to the {@link features/autosave Autosave} guide for details.
 
-### Getting the editor data with `getData()`
+## Initializing the editor with data
 
-If the editor content needs to be retrieved for any reason, like for sending it to the server through an Ajax call, use the `getData()` method:
+By default, the editor is initialized with the content of the element on which it is initialized:
```
Comment (godai78) [inline docs/getting-started/setup/getting-and-setting-data.md:21]: Not sure what this would me, exactly.

**Review by godai78 — APPROVED** [review]

File: docs/getting-started/setup/getting-and-setting-data.md [hunk 2]
```diff
@@ -8,40 +8,60 @@ order: 10
 
 # Getting and setting data
 
-CKEditor 5 allows you to retrieve the data from and save it to your server (or to your system in general) in various ways. In this guide, you can learn about the available options along with their pros and cons.
+CKEditor&nbsp;5 allows you to retrieve the data from and save it to your server (or to your system in general) in various ways. In this guide, you can learn about the available options along with their pros and cons.
```
Comment (filipsobol) [inline docs/getting-started/setup/getting-and-setting-data.md:11]:
```suggestion
In CKEditor&nbsp;5 there are several ways to load and save data. In this guide, you can learn about the available options and their advantages and disadvantages.
```

The original sentence suggested, that it is the CKEditor that handles retrieving and saving the data to and from the database/backend, but that's not always the case (e.g. if someone doesn't use our services).

File: docs/getting-started/setup/getting-and-setting-data.md [hunk 3]
```diff
@@ -8,40 +8,60 @@ order: 10
...
 
-## Manually retrieving the data
+## Autosaving
 
-When you:
+One of the best ways to enhance the user experience in saving editor data is by providing an autosave functionality. This ensures users do not need to remember to save their data, and it prevents your system from being flooded with save events since the save action will be debounced.
```
Comment (filipsobol) [inline docs/getting-started/setup/getting-and-setting-data.md:15]:
```suggestion
One of the best ways to improve the user experience is to automatically save editor data as it changes. This ensures that users do not have to remember to save their data and prevents work loss.
```

IMO we don't need the information about debouncing, as it is an implementation details and is already mentioned in the linked document.

File: docs/getting-started/setup/getting-and-setting-data.md [hunk 4]
```diff
@@ -8,40 +8,60 @@ order: 10
...
-* Use Ajax requests instead of the classic integration with HTML forms,
-* Implement a single-page application,
-* Use a different editor type than the Classic editor (and hence, cannot use the previous method),
+We provide and autosave feature that automatically saves the data (for example sends it to the server) when needed. Please refer to the {@link features/autosave Autosave} guide for details.
 
-### Getting the editor data with `getData()`
+## Initializing the editor with data
 
-If the editor content needs to be retrieved for any reason, like for sending it to the server through an Ajax call, use the `getData()` method:
+By default, the editor is initialized with the content of the element on which it is initialized:
```
Comment (filipsobol) [inline docs/getting-started/setup/getting-and-setting-data.md:21]:
```suggestion
By default, the editor has the content of the DOM element on which it was initialized on.
```

File: docs/getting-started/setup/getting-and-setting-data.md [hunk 5]
```diff
@@ -8,40 +8,60 @@ order: 10
...
+
+```html
+<div id="editor">
+	<!-- This content will appear in the editor if you initialize with this element. -->
+	<p>Hello, world!</p>
+</div>
+```
+
+If you have the data in the JavaScript layer or cannot alter the HTML, the editor’s API provides the `initialData` configuration property. This allows you to provide the content during the initialization of the editor:
```
Comment (filipsobol) [inline docs/getting-started/setup/getting-and-setting-data.md:30]:
```suggestion
However, if you cannot alter the HTML or you load the data asynchronously using JavaScript, you can use the `initialData` configuration property to set the initial state of the editor.
```

File: docs/getting-started/setup/getting-and-setting-data.md [hunk 6]
```diff
@@ -8,40 +8,60 @@ order: 10
...
 
-### Setting the editor data with `setData()`
+The {@link module:core/editor/editorconfig~EditorConfig.initialData `initialData`} property will initialize the editor with the provided data, overriding the content provided at the HTML level.
 
-To replace the editor content with new data, use the `setData()` method:
+If you are setting up the editor with integrations like {@link getting-started/integrations/react React}, consult the documentation for additional properties provided to initialize the data.
+
+## Getting the editor data with `getData()`
+
+If the editor content needs to be retrieved on demand for any reason, such as for sending it to the server through an Ajax call, use the `getData()` method:
```
Comment (filipsobol) [inline docs/getting-started/setup/getting-and-setting-data.md:49]:
```suggestion
You can use the `getData()` method, if you need to get the editor content on demand, such as for sending it to the server using JavaScript APIs.
```

**Review by filipsobol — COMMENTED** [review]
Minor and optional suggestions

File: docs/getting-started/setup/getting-and-setting-data.md (hunk 1 above)
Comment (Witoso) [reply docs/getting-started/setup/getting-and-setting-data.md]: Fixed by Filip.

**Review by filipsobol — APPROVED** [review]

**Review by godai78 — APPROVED** [review]

**Pull request merged**